server/src/main/java/org/opensearch/action/admin/cluster/remotestore/RemoteStoreService.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.cluster.remotestore;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.admin.cluster.repositories.verify.VerifyRepositoryResponse;
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.metadata.Metadata;
+import org.opensearch.cluster.metadata.RepositoriesMetadata;
+import org.opensearch.cluster.metadata.RepositoryMetadata;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.repositories.RepositoriesService;
+import org.opensearch.repositories.Repository;
+import org.opensearch.repositories.RepositoryMissingException;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.function.Supplier;
+
+/**
+ * Contains all the method needed for a remote store node lifecycle.
+ */
+public class RemoteStoreService {
+
+    private static final Logger logger = LogManager.getLogger(RemoteStoreService.class);
+    private final Supplier<RepositoriesService> repositoriesService;
+    public static final Setting<String> REMOTE_STORE_MIGRATION_SETTING = Setting.simpleString(
+        "remote_store.migration",
+        MigrationTypes.NOT_MIGRATING.value,
+        MigrationTypes::validate,
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
+
+    public enum MigrationTypes {
+        NOT_MIGRATING("not_migrating"),
+        MIGRATING_TO_REMOTE_STORE("migrating_to_remote_store"),
+        MIGRATING_TO_HOT("migrating_to_hot");
```

Comment:
There is no such thing as migration, cluster can run in mixed modes, the right semantic would be CompatibilityMode that should support
1. Allow non-remote store to join only as long as there is atleast one non-remote store node
2. Allow non-remote store to join even if there is no non-remote store node.